.github/workflows/main.yml

```diff
@@ -205,6 +206,27 @@ jobs:
       - run: cargo build --release --manifest-path ${{ runner.temp }}/template-test/Cargo.toml
       - run: ${{ runner.temp }}/template-test/target/release/host
 
+  crates-validator:
+    runs-on: [self-hosted, prod, cpu, "${{ matrix.os }}"]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [Linux, macOS]
+    env:
+      RISC0_BUILD_LOCKED: 1
+      RUSTC_WRAPPER: sccache
+    steps:
+      # This is a workaround from: https://github.com/actions/checkout/issues/590#issuecomment-970586842
+      - run: "git checkout -f $(git -c user.name=x -c user.email=x@x commit-tree $(git hash-object -t tree /dev/null) < /dev/null) || :"
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/rustup
+      - uses: ./.github/actions/sccache
+      - run: cargo run --bin cargo-risczero -- risczero install
+        env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: cargo check --manifest-path tools/crates-validator/Cargo.toml
+      - run: cargo test --manifest-path tools/crates-validator/Cargo.toml
```

Comment:
Small note, you could drop `cargo check` in favor of clippy by reusing this block:
https://github.com/risc0/risc0/blob/main/.github/workflows/main.yml#L106C1-L111C1

Reply:
Would it be better to add a new entry in the `test` block, or adding the new block `crates-validator` would be the more appropriate? 

[using it in the existing `test` block](https://github.com/risc0/risc0/pull/903/commits/3d3a6145bca2c027e9dec4beaa8e8590abbda5fe)

Reply:
I kinda like having it as its own `crates-validator` block because then we could possibly disable if that path is not touched with: https://github.com/risc0/paths-filter

Reply:
I changed it to only run `crates-validator` upon changes. hopefully I got `paths-filter`'s documentation right :)